src/allomancy/java/leaf/cosmere/allomancy/common/manifestation/AllomancyZinc.java

```diff
@@ -29,14 +32,74 @@ public AllomancyZinc(Metals.MetalType metalType)
 	protected void applyEffectTick(ISpiritweb data)
 	{
 		int mode = getMode(data);
-
 		String uuid = data.getLiving().getStringUUID();
-		if (mode > 0 && !playerThreadMap.containsKey(uuid))
+
+		// data thread management
 		{
```

Comment:
Same as bendalloy comments, we should probably override tick, and stick this stuff in there. After that, we can have it call the parent class tick